app/src/main/java/org/wikipedia/views/FaceAndColorDetectImageView.java

```diff
@@ -1,29 +1,45 @@
 package org.wikipedia.views;
 
 import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Matrix;
+import android.graphics.Paint;
 import android.graphics.PointF;
+import android.graphics.Rect;
+import android.media.FaceDetector;
 import android.net.Uri;
 import android.util.AttributeSet;
 
 import androidx.annotation.ColorInt;
 import androidx.annotation.DrawableRes;
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
+import androidx.appcompat.widget.AppCompatImageView;
+import androidx.core.content.ContextCompat;
+import androidx.palette.graphics.Palette;
 
-import com.facebook.common.references.CloseableReference;
-import com.facebook.datasource.RetainingDataSourceSupplier;
-import com.facebook.drawee.backends.pipeline.Fresco;
-import com.facebook.drawee.view.SimpleDraweeView;
-import com.facebook.imagepipeline.image.CloseableImage;
-import com.facebook.imagepipeline.request.ImageRequest;
-import com.facebook.imagepipeline.request.ImageRequestBuilder;
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.load.engine.bitmap_recycle.BitmapPool;
+import com.bumptech.glide.load.resource.bitmap.BitmapTransformation;
+import com.bumptech.glide.load.resource.bitmap.DownsampleStrategy;
+import com.bumptech.glide.load.resource.bitmap.TransformationUtils;
 
 import org.wikipedia.R;
+import org.wikipedia.WikipediaApp;
+import org.wikipedia.settings.Prefs;
+import org.wikipedia.util.MathUtil;
+import org.wikipedia.util.log.L;
+
+import java.security.MessageDigest;
 
 import static org.wikipedia.settings.Prefs.isImageDownloadEnabled;
 
-public class FaceAndColorDetectImageView extends SimpleDraweeView {
-    private RetainingDataSourceSupplier<CloseableReference<CloseableImage>> supplier;
+public class FaceAndColorDetectImageView extends AppCompatImageView {
+    public static final int PAINT_FLAGS = Paint.DITHER_FLAG | Paint.FILTER_BITMAP_FLAG;
+    private static final Paint DEFAULT_PAINT = new Paint(PAINT_FLAGS);
+    private static final int BITMAP_COPY_WIDTH = 200;
 
     public interface OnImageLoadListener {
```

Comment:
It is not showing here but should we remove the unused `onImageFailed()` method from the interface?

Reply:
ok!